driver-lambda/src/main/com/mongodb/lambdatest/LambdaTestApp.java

```diff
@@ -0,0 +1,143 @@
+    private double totalCommandDuration = 0;
+
+    public LambdaTestApp() {
+        String connectionString = System.getenv("MONGODB_URI");
+
+        MongoClientSettings settings = MongoClientSettings.builder()
+                .applyConnectionString(new ConnectionString(connectionString))
+                .addCommandListener(new CommandListener() {
+                    @Override
+                    public void commandSucceeded(@NonNull final CommandSucceededEvent event) {
+                        totalCommandCount++;
+                        totalCommandDuration += event.getElapsedTime(MILLISECONDS);
+                    }
+                    @Override
+                    public void commandFailed(@NonNull final CommandFailedEvent event) {
+                        totalCommandCount++;
+                        totalCommandDuration += event.getElapsedTime(MILLISECONDS);
+                    }
+                })
+                .applyToServerSettings(builder -> builder.addServerMonitorListener(new ServerMonitorListener() {
+                    @Override
+                    public void serverHeartbeatSucceeded(@NonNull final ServerHeartbeatSucceededEvent event) {
+                        totalHeartbeatCount++;
+                        totalHeartbeatDuration += event.getElapsedTime(MILLISECONDS);
+                    }
+                    @Override
+                    public void serverHeartbeatFailed(@NonNull final ServerHeartbeatFailedEvent event) {
+                        totalHeartbeatCount++;
+                        totalHeartbeatDuration += event.getElapsedTime(MILLISECONDS);
+                    }
+                }))
+                .applyToConnectionPoolSettings(builder -> builder.addConnectionPoolListener(new ConnectionPoolListener() {
+                    @Override
+                    public void connectionCreated(@NonNull final ConnectionCreatedEvent event) {
+                        openConnections++;
+                    }
+                    @Override
+                    public void connectionClosed(@NonNull final ConnectionClosedEvent event) {
+                        openConnections--;
+                    }
+                }))
+                .build();
+        mongoClient = MongoClients.create(settings);
+    }
+
+    public APIGatewayProxyResponseEvent handleRequest(final APIGatewayProxyRequestEvent input, final Context context) {
+        try {
+            MongoCollection<Document> collection = mongoClient
+                    .getDatabase("lambdaTest")
+                    .getCollection("test");
+            BsonValue id = collection.insertOne(new Document("n", 1)).getInsertedId();
+            collection.deleteOne(new Document("_id", id));
+
+            BsonDocument responseBody = new BsonDocument()
+                    .append("totalCommandDuration", new BsonDouble(totalCommandDuration))
+                    .append("totalCommandCount", new BsonDouble(totalCommandCount))
```

Comment:
I might be missing something, however, as I see spec requires us only to count the durations of the commands but not the total count of commands.

> Drivers MUST record the durations and counts of the heartbeats, the durations of the commands, as well as keep track of the number of open connections, and report this information in the function response as JSON.

Reply:
I agree, but I do not think it will be harmful to provide the count as well for consistency; also, Node supplies this as well.